### ISSUE TITLE
Add translation guide

### ISSUE THREAD

**Issue by DominikNovosel** [wonderunit/storyboarder#2330]
Add a wiki page that details special procedures for localizing the app.

**Comment by audionerd** [comment]
Are you interested in making a translation? 

Use the Language Editor (in Preferences) to start the process
The result can be exported as a JSON file
Submit that JSON file in a PR (eg. for US English, the file is `src/js/locales/en-US.json`) and we'll review

Would you want to start some notes in this thread on what you discover in the translation process, and we could turn it into a wiki page eventually?

Here's the current US English JSON (although this will change slightly once the `pdf-export` branch is merged):
https://github.com/wonderunit/storyboarder/blob/master/src/js/locales/en-US.json

**Comment by audionerd** [comment]
Here's an example PR of the process https://github.com/wonderunit/storyboarder/pull/2125

**Comment by DominikNovosel** [comment]
Hi, thanks for responding! Making a translation wasn't my original intent, but now that I think about it, some of my colleagues might benefit from a localized version of the app. I'll ask around to see whether this would be worth doing since almost everyone that would use this app in my country is fluent enough in English to use it regardless, not to mention our woeful lack of widely accepted technical terminology. If I do get around to translating this, I'll be sure to share my discoveries to improve the documentation (Which, I beg your pardon, looks a mess at the moment. Should I open another issue for this?).

Quick question, does Storyboarder support language-specific plurals? Some languages require special treatment of words associated with specific numbers (eg. in Croatian: 0-1 godin**a**, 2-4 godin**e**, 5 godin**a**...). You sent me Chinese as an example, which notably has no such rules. You can see more examples here: https://developer.android.com/guide/topics/resources/string-resource#Plurals

**Comment by audionerd** [comment]
> the documentation (Which, I beg your pardon, looks a mess at the moment. Should I open another issue for this?)
lol yes it is. identifying specific under-documented issues (i was trying to do x and couldn't figure it out anywhere in the app, it should be described somewhere and the steps are y) could be very helpful. a generic catch-all open issue re documentation would not be very helpful however (we know it's a problem).
it all cases, it would be preferable to make outside documentation un-necessary by improving the app or providing explanatory context directly within it.

> does Storyboarder support language-specific plurals?
Good question. We use i18next, which has that capability, but I don't think we take advantage of it.
https://www.i18next.com/translation-function/plurals

**Comment by DominikNovosel** [comment]
That's quite problematic for some languages, but I can't bring myself to blame you, the plural syntax is JANKY and a pain to look at. 🤣

Btw, I had an idea in issue #1663 and it's just crazy enough to work for... something, and I'd like to hear what you think.

**Comment by DominikNovosel** [comment]
Just finished the Croatian translation. https://github.com/DominikNovosel/Storyboarder-hr_HR
In hindsight, the process wasn't particularly notable, other than the fact that a lot of strings seemed to be missing, and some that were present in the JSON wouldn't show up in the UI.
